android/quest/src/main/java/org/smartregister/fhircore/quest/ui/questionnaire/QuestionnaireViewModel.kt

```diff
@@ -820,6 +821,28 @@ constructor(
     }
   }
 
+  /**
+   * This function recursively fetches and loads all dependent structure maps as defined here
+   * https://hl7.org/fhir/R4B/structuremap-definitions.html#StructureMap.import to the worker
+   * context cache
+   *
+   * @param structureMapId String id of the parent StructureMap
+   * @return [StructureMap] the parent StructureMap resource
+   */
+  private suspend fun loadStructureMapDependencyTree(
+    structureMapId: String,
+  ): StructureMap? {
+    val structureMap = defaultRepository.loadResource<StructureMap>(structureMapId)
+    structureMap?.also { structureMapIterator ->
+      simpleWorkerContext.cacheResource(structureMapIterator)
+
+      structureMapIterator.import?.let { canonicalTypes ->
+        canonicalTypes.forEach { loadStructureMapDependencyTree(it.value) }
```

Comment:
we need to filter out the id here

```suggestion
        canonicalTypes.forEach { structureMapUrl ->
          structureMapUrl.value?.substringAfterLast("/")?.let { structureMapId ->
            loadStructureMapDependencyTree(structureMapId)
          }
        }
```